### ISSUE TITLE
TravisCI Build Halts After Tests Finish

### ISSUE THREAD

**Issue by pkkid** [pkkid/python-plexapi#137]
This is happening quite frequently.  We currently just go into Travis and tell it to rerun the build and it will pass on a second try. We should fix this.

Good news: I was able to reproduce this locally finally. It looks like a thread-lock issue when py.test completes. Idiot me hit `ctrl+c` to kill it before realizing it was this issue that we hit, so I'll dig in more next time it happens.

**Comment by Hellowlol** [comment]
That hasnt happend when it test locally, not even once. I dont get it. We dont use any threads except websocket.

**Comment by pkkid** [comment]
The stacktrace on the screen with py.test waiting for a lockfile.  I wish I realized what I was looking at before I killed the process and restarted the test. Then I would have captured more information.  I tried to make it happen locally again for an hour before bed with no success.

I just added the `-s` option to pytest in travis which means "don't capture stdout".  Maybe it will help, but I'm basically throwing things against the wall to see what sticks at this point.

**Comment by Hellowlol** [comment]
Hehe, let's hope something sticks then. Is it doesn't help I would like to disable it later on.

**Comment by pkkid** [comment]
I think this is the issue:

```python
Exception ignored in: <module 'threading' from '/usr/lib/python3.5/threading.py'>
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 1288, in _shutdown
    t.join()
  File "/usr/lib/python3.5/threading.py", line 1054, in join
    self._wait_for_tstate_lock()
  File "/usr/lib/python3.5/threading.py", line 1070, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
KeyboardInterrupt```

**Comment by pkkid** [comment]
Another attempt at fixing this. I noticed a few things:

1. There didn't see reference to threading anywhere in the pytest source as I thought there may be, so that theory is out the window.

2. I noticed our threading utility function was not declaring its threads as daemon threads.  Meaning they were not being killed if stuck open for some odd reason when the main process ends. I added this in. Fingers crossed!
